internal/controllers/topology/cluster/scope/state.go

```diff
@@ -123,3 +127,49 @@ func (md *MachineDeploymentState) IsUpgrading(ctx context.Context, c client.Clie
 	}
 	return false, nil
 }
+
+// MachinePoolsStateMap holds a collection of MachinePool states.
```

Comment:
Please compare the new MachinePool part here with MachineDeployment above. I think we refactored the code above in parallel to you writing the new one below

Also please try to use the same order. I'm just comparing MD with MP to spot potential errors and the same ordering makes that a lot easier

Reply:
Will do! This part was written by Richard and I thought it was something specific to MachinePools, but it looks like the code was refactored since he worked on it last.

Reply:
Done, let me know if it looks good!

Reply:
Look good, will open new comments if necessary